### PR TITLE
Separate out tests to cater for changes in Python 3.8.8

### PR DESCRIPTION
**Pull request by kaxil**
https://github.com/python/cpython/pull/24297 change was included in
Python 3.8.8 to fix a vulnerability (bpo-42967)

Depending on which Base Python Image is run in our CI, two of the tests
can fail or succeed.

Our Previous two attempts:

- https://github.com/apache/airflow/commit/061cd236deb22567e4de36af11025f028d787989#
- https://github.com/apache/airflow/commit/49952e79b04da932242ebf3981883e591b467994

We might for a while get different base python version depending on the changes of a PR (whether or not it includes a change to dockerfiler).
a) when you have PR which do not have changes in the Dockerfile, they will use the older python version as base (for example Python 3.8.7)
b) when you have PR that touches the Dockerfile and have setup.py changes in master, it should pull Python 3.8.8 first.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: tests/www/test_views.py [hunk 1]
```diff
@@ -2784,7 +2784,38 @@ def test_trigger_dag_form(self):
             ("%2Fgraph%3Fdag_id%3Dexample_bash_operator", "/graph?dag_id=example_bash_operator"),
         ]
     )
-    def test_trigger_dag_form_origin_url(self, test_origin, expected_origin):
+    @pytest.mark.skipif(
+        sys.version_info < (3, 8, 8),
```
Comment (kaxil) [inline tests/www/test_views.py:2788]: I did not do:

```python
sys.version_info <= (3, 8, 7)
```

for the following case:

```python
❯ python
Python 3.7.4 (default, Aug 13 2019, 15:17:50)
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.version_info == (3, 7, 4, )
False
>>> sys.version_info
sys.version_info(major=3, minor=7, micro=4, releaselevel='final', serial=0)
>>> sys.version_info == (3, 7, 4, 'final', 0)
True
```

File: tests/www/test_views.py [hunk 2]
```diff
@@ -3330,7 +3361,36 @@ class TestHelperFunctions(TestBase):
...
+            assert get_safe_url(test_url) == expected_url
+
+    @parameterized.expand(
+        [
+            ("", "/home"),
+            ("http://google.com", "/home"),
+            (
+                "http://localhost:8080/trigger?dag_id=test_dag&origin=%2Ftree%3Fdag_id%test_dag';alert(33)//",
+                "http://localhost:8080/trigger?dag_id=test_dag&origin=%2Ftree%3F"
+                "dag_id%25test_dag%27&alert%2833%29%2F%2F=",
```
Comment (XD-DENG) [inline tests/www/test_views.py]: This line is missed to be updated?

**Comment by ashb** [comment]
> Depending on which Base Python Image is run in our CI, two of the tests

You mean this isn't predictable?

**Comment by kaxil** [comment]
> > Depending on which Base Python Image is run in our CI, two of the tests
> 
> You mean this isn't predictable?

Yup, from https://github.com/apache/airflow/blob/6851677a89294698cbdf9fa559bf9d12983c88e0/scripts/ci/libraries/_push_pull_remove_images.sh#L107-L146

Jarek has explained and he is going to fix that in a follow-up PR

**Review by vikramkoka — APPROVED** [review]

**Comment by potiuk** [comment]
Nice! Thanks @kaxil !

**Pull request merged**